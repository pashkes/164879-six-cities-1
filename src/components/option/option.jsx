import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {ActionCreators} from "./../../reducer/data/data";

export class Option extends PureComponent {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(evt) {
    evt.preventDefault();
    this.props.onSelected(this.props.value, this.props.name);
    this.props.typeSort(this.props.value);
  }

  render() {
    const {isSelected, value, name} = this.props;
    return (
      <li className={`${isSelected ? `places__option--active` : ``} places__option`} onClick={this.onClick} tabIndex="0" data-value={value}>{name}</li>
    );
  }
}

Option.propTypes = {
  onSelected: PropTypes.func.isRequired,
  isSelected: PropTypes.bool.isRequired,
  value: PropTypes.string.isRequired,
  typeSort: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  typeSort: (value) => dispatch(ActionCreators.changeSortType(value)),
});

export default connect(null, mapDispatchToProps)(Option);
