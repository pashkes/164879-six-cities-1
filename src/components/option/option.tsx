import * as React from "react";
import {connect} from "react-redux";

import {ActionCreators} from "./../../reducer/data/data";
import {KeyCode} from "./../../constants";

interface Props {
  value: string,
  name: string,
  isSelected: boolean,
  onSetTypeSort: (value: string) => void,
  onSelectOption: (value: string, name: string) => void,
  onFocusButton: () => void,
}

export class Option extends React.PureComponent<Props, null> {
  handleOptionClick = () => {
    const {
      value,
      name,
      onSetTypeSort,
      onSelectOption,
      onFocusButton,
    } = this.props;

    onSelectOption(value, name);
    onSetTypeSort(value);
    onFocusButton();
  };

  handleKeyDownEnter = (evt) => {
    const {
      onFocusButton,
      name,
      value,
      onSelectOption,
      onSetTypeSort,
    } = this.props;

    if (evt.keyCode === KeyCode.ENTER) {
      onSelectOption(value, name);
      onSetTypeSort(value);
      onFocusButton();
    }
  };

  render() {
    const {isSelected, value, name} = this.props;

    return (
      <li
        className={`${isSelected ? `places__option--active` : ``} places__option`}
        onKeyDown={this.handleKeyDownEnter}
        onClick={this.handleOptionClick}
        tabIndex={0}
        data-value={value}
      >
        {name}
      </li>
    );
  }
}


const mapDispatchToProps = (dispatch) => ({
  onSetTypeSort: (value) => dispatch(ActionCreators.setSortType(value)),
});

export default connect(null, mapDispatchToProps)(Option);
