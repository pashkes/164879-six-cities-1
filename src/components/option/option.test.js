import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import {Option} from './option.jsx';
configure({adapter: new Adapter()});

it(`Option renders correctly`, () => {
  const option = shallow(<Option name={`Popular`} typeSort={jest.fn()} onSelected={jest.fn()} value={``} isSelected={true} />);
  expect(option).toMatchSnapshot();
});
