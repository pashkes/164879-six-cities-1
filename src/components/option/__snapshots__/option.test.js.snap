// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option renders correctly 1`] = `
<li
  className="places__option--active places__option"
  data-value=""
  onClick={[Function]}
  tabIndex="0"
>
  Popular
</li>
`;
